src/sinks/datadog/metrics/config.rs

```diff
@@ -16,29 +16,35 @@ use crate::{
     http::HttpClient,
     sinks::{
         datadog::DatadogCommonConfig,
-        util::{batch::BatchConfig, ServiceBuilderExt, SinkBatchSettings, TowerRequestConfig},
+        util::{
+            batch::BatchConfig, Concurrency, ServiceBuilderExt, SinkBatchSettings,
+            TowerRequestConfig,
+        },
         Healthcheck, UriParseSnafu, VectorSink,
     },
     tls::{MaybeTlsSettings, TlsEnableableConfig},
 };
 
-// This default is centered around "series" data, which should be the lion's share of what we
-// process.  Given that a single series, when encoded, is in the 150-300 byte range, we can fit a
-// lot of these into a single request, something like 150-200K series.  Simply to be a little more
-// conservative, though, we use 100K here.  This will also get a little more tricky when it comes to
-// distributions and sketches, but we're going to have to implement incremental encoding to handle
-// "we've exceeded our maximum payload size, split this batch" scenarios anyways.
-pub const MAXIMUM_PAYLOAD_COMPRESSED_SIZE: usize = 3_200_000;
-pub const MAXIMUM_PAYLOAD_SIZE: usize = 62_914_560;
-
 // TODO: revisit our concurrency and batching defaults
 const DEFAULT_REQUEST_RETRY_ATTEMPTS: usize = 5;
 
+const DEFAULT_REQUEST_FIXED_CONCURRENCY: usize = 150;
+
 #[derive(Clone, Copy, Debug, Default)]
 pub struct DatadogMetricsDefaultBatchSettings;
 
+// This default is centered around "series" data, which should be the lion's share of what we
+// process. The value of 1k events was arrived at empirically by testing the sink's performance
+// with the v2 series endpoint.
 impl SinkBatchSettings for DatadogMetricsDefaultBatchSettings {
-    const MAX_EVENTS: Option<usize> = Some(100_000);
+    // TODO: the series V1 endpoint support is considered deprecated and should be removed in a future release.
+    // At that time when the V1 support is removed, the 1k value for max events can be used directly.
+    const MAX_EVENTS: Option<usize> = Some(
+        match option_env!("VECTOR_TEMP_USE_DD_METRICS_SERIES_V1_API") {
+            Some(_) => 100_000,
+            None => 1_000,
+        },
```

Comment:
Does it make sense to use `map_or` here or does this just obfuscate what's going on?
```suggestion
        option_env!("VECTOR_TEMP_USE_DD_METRICS_SERIES_V1_API").map_or(1_000, |_| 100_000),
```

Reply:
I would argue it lands on the side of obfuscation but also will become irrelevant in the next diff.